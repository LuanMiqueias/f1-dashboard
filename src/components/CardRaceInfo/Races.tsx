import CardRaces from "../CardRaces/CardRaces";
import RaceInfo from "../RaceInfo/RaceInfo";
import styles from "./style.module.css";

interface PropsRaces {
  id: number;
  distance?: string;
  date: string;
  status: string;
  season: number;
  type: string;
  laps: {
    total?: number;
  };
  circuit: {
    id: number;
    name: string;
    image: string;
  };
}
interface IData {
  data: PropsRaces[];
}
const Races = ({ data }: IData) => {
  return (
    <div className={styles.container}>
      <div className={styles.content_races}>
        {data.map((item) => {
          return (
            <CardRaces
              key={item.id}
              id={item.id}
              circuit={item.circuit}
              type={item.type}
              date={item.date}
            />
          );
        })}
      </div>
      <div className={styles.content_races}>
        <RaceInfo />
      </div>
    </div>
  );
};

export default Races;
