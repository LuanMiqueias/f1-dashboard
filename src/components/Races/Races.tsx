import React from "react";
import { RacesContext } from "../../context/ContextRaces";
import CardRaces from "../CardRaces/CardRaces";
import Loading from "../Loading/Loading";
import RaceInfo from "../RaceInfo/RaceInfo";
import styles from "./style.module.css";

interface PropsRaces {
  id: number;
  distance?: string;
  date: string;
  status: string;
  season: number;
  type: string;
  laps: {
    total?: number;
  };
  circuit: {
    id: number;
    name: string;
    image: string;
  };
}
interface IData {
  data: PropsRaces[];
}
const Races = ({ data }: IData) => {
  const { loading } = React.useContext(RacesContext);
  return (
    <div className={styles.container}>
      <div className={styles.content_races}>
        {loading ? (
          <Loading />
        ) : (
          data.map((item) => {
            return (
              <CardRaces
                key={item.id}
                id={item.id}
                circuit={item.circuit}
                type={item.type}
                date={item.date}
              />
            );
          })
        )}
      </div>
      {screen.width >= 960 && (
        <div className={`${styles.content_races} ${styles.content_races_info}`}>
          <RaceInfo />
        </div>
      )}
    </div>
  );
};

export default Races;
