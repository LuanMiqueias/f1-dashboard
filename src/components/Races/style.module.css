.container{
  display: flex;
  align-items: stretch;
  justify-content: flex-start;
  flex-wrap: wrap;
  gap:16px;
  animation: animaShow .2s ease-out forwards;
}
.content_races{
  border-radius: 5px;
  flex:1;
  display: flex;
  flex-direction: column;
  gap:8px;
}
.content_races_info{
  padding: 0rem;
  flex:1.5;
  width: 100%;
  min-width: 250px;
  max-height: 400px;
  background: var(--gray-light);
  border:solid 2px #eee;
  position: sticky;
  top:32px;
}
@keyframes animaShow {
  0%{
    opacity: 0;
    transform: scale(0.9);
  }
  50%{
    opacity: initial;
  }
  100%{
      transform: initial;
  }
}