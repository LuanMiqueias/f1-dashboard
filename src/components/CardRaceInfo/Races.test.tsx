import { render, screen } from "@testing-library/react";
import Races from "./Races";

const fakeData = [
  {
    id: 1,
    date: "1",
    status: "test",
    season: 2021,
    type: "string",
    laps: {
      total: 10,
    },
    circuit: {
      id: 2,
      name: "string-test",
      image: "img",
    },
  },
];
it("Should math snapshot", () => {
  const { container } = render(<Races data={fakeData} />);

  expect(container.firstChild).toMatchSnapshot();
});
