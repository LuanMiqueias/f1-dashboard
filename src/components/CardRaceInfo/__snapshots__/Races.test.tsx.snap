// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should math snapshot 1`] = `
<div
  class="container"
>
  <div
    class="content_races"
  >
    <div
      class="card_races"
    >
      <div
        class="content_card_races"
      >
        <h2>
          string-test
        </h2>
        <p>
          string
        </p>
      </div>
      <div
        class="content_card_races"
      >
        <p>
          Jan 1, 00:00 PM
        </p>
      </div>
    </div>
  </div>
  <div
    class="content_races"
  >
    <div
      class="container"
    >
      <div
        class="content"
        style="min-height: 400px;"
      >
        <p
          class="text"
        >
          Select one race for more informations
        </p>
      </div>
    </div>
  </div>
</div>
`;
